src/beautifiers/eslint.coffee

```diff
@@ -23,12 +24,25 @@ module.exports = class ESLintFixer extends Beautifier
       allowUnsafeNewFunction ->
         importPath = Path.join(projectPath, 'node_modules', 'eslint')
         try
-          CLIEngine = require(importPath).CLIEngine
+          context.which('eslint').then((globalEslintPath) ->
+            if globalEslintPath != 'eslint'
+              importPath = Path.resolve(globalEslintPath, '../../lib/node_modules/eslint')
+            else
+              needInstallMsg = 'Error: it seems like you did not install eslint globally,'
+              needInstallMsg += 'try to run npm i -g eslint and restart the atom editor.'
+              context.deprecate needInstallMsg
+              resolve undefined
+            CLIEngine = require(importPath).CLIEngine
 
-          cli = new CLIEngine(fix: true, cwd: projectPath)
-          result = cli.executeOnText(text).results[0]
-
-          resolve result.output
+            cli = new CLIEngine(fix: true, cwd: projectPath)
+            result = cli.executeOnText(text).results[0]
+            if not result.output
+              errMsg = result.messages[0].message
+              errLine = result.messages[0].line
+              errColumn = result.messages[0].column
+              context.deprecate 'Error: [' + errMsg + '] at line ' + errLine + ', column ' + errColumn
```

Comment:
`deprecate` does not appear to be applicable here. Please `reject(new Error("insert message here"))` and/or use the logger methods (e.g. `      @verbose("insert message here")`)

Reply:
What does `result.messages[0]` look like? If it is an `Error` instance than please simply:

```coffeescript
error = result.messages[0]
return reject(error)
```